parameters.cpp

```diff
@@ -553,67 +554,49 @@ void Parameters::getParameters() {
-         if (maxNumPassesPtr != numPasses.end()) {
-            maxNumPassesInt = *maxNumPassesPtr;
-         } else {
-            cerr << "Trying to dereference null pointer \t"
-                 << " in " << __FILE__ << ":" << __LINE__ << endl;
+   // If we are in an AMR run we need to set up the filtering scheme.
+   if (P::amrMaxSpatialRefLevel>0){
+      bool isEmpty = blurPassString.size() == 0;
+      if (!isEmpty){
+         //sanity check=> user should define a pass for every level
+         if (blurPassString.size() != P::amrMaxSpatialRefLevel + 1) {
+            cerr << "Filter Passes=" << blurPassString.size() << "\t" << "AMR Levels=" << P::amrMaxSpatialRefLevel + 1 << endl;
+            cerr << "FilterPasses do not match AMR levels. \t" << " in " << __FILE__ << ":" << __LINE__ << endl;
             MPI_Abort(MPI_COMM_WORLD, 1);
          }
-
-         printf("Filtering is on with max number of Passes= \t%d\n", maxNumPassesInt);
-         int lev = 0;
-         for (auto& iter : P::numPasses) {
-            printf("Refinement Level %d-->%d Passes\n", lev, iter);
-            lev++;
+         //sort the filtering passes per refLevel
+         numPasses.clear();
+         std::sort(blurPassString.begin(), blurPassString.end(),
+                  [](const std::string& lhs, const std::string& rhs)->bool 
+                  { return (lhs.size() < rhs.size()) || (lhs.size() == rhs.size() && lhs < rhs);});
+
+         int maxPasses=stoi(blurPassString.back());
+         std::vector<std::string>::const_iterator it = blurPassString.end();
+         for (uint refLevel=0; refLevel<P::amrMaxSpatialRefLevel+1; refLevel++){
+            numPasses[refLevel] =stoi(*(--it));
          }
-      }
-   } else {
-      numPasses = {0};
-
-      if (myRank == MASTER_RANK) {
-         maxNumPassesPtr = max_element(P::numPasses.begin(), P::numPasses.end());
-         if (maxNumPassesPtr != numPasses.end()) {
-            maxNumPassesInt = *maxNumPassesPtr;
-         } else {
-            cerr << "Trying to dereference null pointer \t"
-                 << " in " << __FILE__ << ":" << __LINE__ << endl;
-            MPI_Abort(MPI_COMM_WORLD, 1);
+      }else{
+         //here we will default to manually constructing the number of passes
+         numPasses.clear();
+         auto g_sequence=[](int size){
+            int retval=1;
+            while(size!=0){
+               retval*=2;
+               size-=1;
+            }
+            return retval;
+         };
+         int maxPasses=g_sequence(P::amrMaxSpatialRefLevel-1);
+         for (uint refLevel=0; refLevel<=P::amrMaxSpatialRefLevel; refLevel++){
+            numPasses[refLevel] =maxPasses;
+            maxPasses/=2; 
          }
-
-         printf("Filtering is off and max number of Passes is = \t %d\n",
-                *max_element(P::numPasses.begin(), P::numPasses.end()));
       }
+      P::maxFilteringPasses = numPasses[0];
```

Comment:
I think it would be better to construct these looping from uint refLevel=P::amrMaxSpatialRefLevel with refLevel-- and doing maxPasses*=2 each loop.